vars/nexusUpload.groovy

```diff
@@ -0,0 +1,90 @@
+import static com.sap.piper.Prerequisites.checkScript
+
+import static groovy.json.JsonOutput.toJson
+
+import com.sap.piper.PiperGoUtils
+
+
+import com.sap.piper.Utils
+
+import groovy.transform.Field
+
+@Field String METADATA_FILE = 'metadata/nexusUpload.yaml'
+@Field String STEP_NAME = getClass().getName()
+
+
+void call(Map parameters = [:]) {
+
+    handlePipelineStepErrors (stepName: STEP_NAME, stepParameters: parameters) {
+
+        final Script script = checkScript(this, parameters) ?: null
+
+        if (!script) {
+            error "Reference to surrounding pipeline script not provided (script: this)."
+        }
+
+        def utils = parameters.juStabUtils ?: new Utils()
+        parameters.juStabUtils = null
+
+        if (!parameters.get('credentialsId')) {
+            // Remove null or empty credentialsId key. (Eases calling code.)
+            parameters.remove('credentialsId')
+        }
+
+        echo "nexusUpload parameters: $parameters"
+
+        sh 'git clone https://github.com/SAP/jenkins-library.git -b nexus-upload .piper-git-checkout'
+        dir('.piper-git-checkout') {
+            dockerExecute(script: this, dockerImage: 'golang:1.13', dockerOptions: '-u 0') {
```

Comment:
[clarification] I guess this is preliminary coding during the development phase which will be replaced by the commented code below before merging

Reply:
Yes, exactly. This will then be replaced again by the commented code (PiperUtils unstashing the piper binary).